src/main.cpp

```diff
@@ -14,4 +15,13 @@ int main() {
     // Literal syntax
     auto a1 = "a1"_sq;
     uint64_t board = 1ULL << static_cast<int>(a1);
+
+    Bitboard myboard;
+    std::cout << myboard << std::endl;
+
+    auto d7 = "d7"_sq;
+    uint64_t board2 = 1ULL << static_cast<int>(d7);
```

Comment:
Why is this `uint64_t`?

Also, what happens if you don't static cast? Does it give a compilation error?

Reply:
`Bitboard` constructor takes `uint64_t` as input

Reply:
No it does not give compilation error. It is useful since we know the datatypes during compilation

Reply:
> `Bitboard` constructor takes `uint64_t` as input

The type of `myboard2` should be `Bitboard`.

> No it does not give compilation error.

That's useful to know.

Reply:
done